browser/request_otr/request_otr_browsertest.cc

```diff
@@ -507,4 +517,25 @@ IN_PROC_BROWSER_TEST_F(RequestOTRServiceWorkerBrowserTest,
   ASSERT_TRUE(content::ExecJs(web_contents(), "setup();"));
 }
 
+// Define a subclass that sets up a special HTTP server that responds with
+// a custom header to trigger an OTR tab.
+class RequestOTRCustomHeaderBrowserTest : public RequestOTRBrowserTest {
+ public:
+  void SetUp() override {
+    content::SetupCrossSiteRedirector(embedded_test_server());
+    embedded_test_server()->RegisterRequestHandler(
+        base::BindRepeating(&RespondWithCustomHeader));
+    ASSERT_TRUE(embedded_test_server()->Start());
+    ExtensionBrowserTest::SetUp();
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(RequestOTRCustomHeaderBrowserTest,
```

Comment:
I think we also need to check that it doesn't show when the header is not present? Of course other tests would break if it always showed up, but just for completeness here